comp/forwarder/defaultforwarder/resolver/domain_resolver.go

```diff
@@ -33,38 +37,98 @@ type DomainResolver interface {
 	// Resolve returns the domain to be used to send data for a given `transaction.Endpoint` along with a
 	// destination type
 	Resolve(endpoint transaction.Endpoint) (string, DestinationType)
+	// GetEndpoints returns the list of endpoints associated with this `DomainResolver`
+	GetEndpoints() []utils.APIKeys
 	// GetAPIKeys returns the list of API Keys associated with this `DomainResolver`
 	GetAPIKeys() []string
+	// SetAPIKeys updates the deduped list of API keys associated with this `DomainResolver`
+	SetAPIKeys(keys []string)
 	// GetBaseDomain returns the base domain for this `DomainResolver`
 	GetBaseDomain() string
 	// GetAlternateDomains returns all the domains that can be returned by `Resolve()` minus the base domain
 	GetAlternateDomains() []string
 	// SetBaseDomain sets the base domain to a new value
 	SetBaseDomain(domain string)
 	// UpdateAPIKey replaces instances of the oldKey with the newKey
-	UpdateAPIKey(oldKey, newKey string)
+	UpdateAPIKey(configPath, oldKey, newKey string)
 	// GetBearerAuthToken returns Bearer authtoken, used for internal communication
 	GetBearerAuthToken() string
 }
 
 // SingleDomainResolver will always return the same host
 type SingleDomainResolver struct {
-	domain  string
-	apiKeys []string
-	mu      sync.Mutex
+	domain         string
+	apiKeys        []utils.APIKeys
+	dedupedAPIKeys []string
+	mu             sync.Mutex
+}
+
+// OnUpdateConfig adds a hook into the config which will listen for updates to the API keys
+// of the resolver.
+func OnUpdateConfig(resolver DomainResolver, log log.Component, config config.Component) {
+	config.OnUpdate(func(setting string, oldValue, newValue any) {
+		found := false
+
+		for _, endpoint := range resolver.GetEndpoints() {
+			if endpoint.ConfigSettingPath == setting {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return
+		}
+
+		if setting == "additional_endpoints" {
+			// Updating additional endpoints don't give us the exact key that has been updated so we reload the whole config section.
```

Comment:
The live process checks also make use of the default forwarder and want to leverage it for api key refreshing, but our `additional_endpoints` config is different (`process_config.additional_endpoints`) so this won't work for us. It seems like we can easily add support here for it just by checking if the setting contains `additional_endpoints` as a substring. Can we do that here? If it out of scope, it can be done in another PR. I would mostly like to know if that approach makes sense.

Reply:
Yes, good catch. I have updated it to check it contains `additional_endpoints`.

Reply:
It still doesn't seem to be working correctly for `process_config`. Am investigating...

Reply:
Ok. This should be good now.